### ISSUE TITLE
v3.0.0: Acoustic scene analysis and audio event detection

### ISSUE THREAD

**Issue by EffortlessSteven** [EffortlessMetrics/slower-whisper#62]
## Summary

Extend audio enrichment to classify acoustic scenes (office, street, home, call center) and detect non-speech audio events (laughter, applause, door slams, typing, coughing).

## Motivation

Current enrichment focuses on speech-only features. Real conversations happen in rich acoustic environments that provide context:
- Background noise affects speech quality and speaker stress
- Non-speech sounds (laughter, sighs) convey emotional state
- Acoustic scene informs appropriate analysis thresholds

### 1. Acoustic Scene Classification
- Indoor: office, home, car, restaurant
- Outdoor: street, park
- Communication: phone call, video conference
- Output: `meta.acoustic_scene: "office"`

### 2. Audio Event Detection
- Human sounds: laughter, cough, sigh, applause
- Environmental: door, phone ring, typing, music
- Output per-segment: `audio_events: [{"type": "laughter", "start": 5.2, "end": 5.8}]`

### 3. Noise Level Metrics
- Background SNR estimate
- Speech/non-speech ratio
- Reverberation indicator

## Architecture

- Use pre-trained YAMNet or PANNs for initial classification
- Fine-tune on conversational audio
- Integrate as optional L2 enrichment pass

## Schema Extension

```json
"meta.acoustic_context": {
  "scene": "office",
  "scene_confidence": 0.72,
  "snr_estimate_db": 15.5
},
"segment.audio_events": [
  {"type": "laughter", "start": 5.2, "end": 5.8, "confidence": 0.88}
]
```

## Prior Art

- Google AudioSet + YAMNet (521 audio event classes)
- Kong et al., "PANNs: Large-Scale Pretrained Audio Neural Networks" (2020)
- DCASE Challenge acoustic scene classification

## Acceptance Criteria

- [ ] Scene classification with >80% accuracy on test set
- [ ] Detection of 10+ common audio events
- [ ] SNR estimation within ±3dB
- [ ] Streaming support for real-time detection
